Reconcile the minikube install page with best practices (#20717)
kubernetes/website: content/en/docs/tasks/tools/install-minikube.md

{{% capture overview %}}

> This page shows you how to install [minikube](https://minikube.sigs.k8s.io/), a tool that runs a single-node Kubernetes cluster on your personal computer.

Reviewer: > a single-node Kubernetes cluster

should we drop the 'single-node'?

{{% /capture %}}

{{% capture prerequisites %}}

To run minikube, you'll need:

> * 2GB of free memory
> * 20GB of free disk space

Reviewer 2: mini nit:
```suggestion
* 2GiB of free memory
* 20GB of free disk space
```

* Internet connection

> * A compatible container or virtual machine manager, such as: [Docker](https://minikube.sigs.k8s.io/docs/docs/drivers/docker), [Hyperkit](https://minikube.sigs.k8s.io/docs/docs/drivers/hyperkit), [Hyper-V](https://minikube.sigs.k8s.io/docs/docs/drivers/hyperv), [KVM](https://minikube.sigs.k8s.io/docs/docs/drivers/kvm2), [Parallels](https://minikube.sigs.k8s.io/docs/docs/drivers/parallels), [Podman](https://minikube.sigs.k8s.io/docs/docs/drivers/podman), [VirtualBox](https://minikube.sigs.k8s.io/docs/docs/drivers/virtualbox), or [VMWare](https://minikube.sigs.k8s.io/docs/docs/drivers/vmware)

Reviewer 2: nit: This looks like it would work better as a Markdown list. Also, I'm not clear if that's intended to seem like an exhaustive list or not.

Author: It isn't entirely exhaustive, but it does include the most inquired about options. I'm OK to narrow the list down, but I felt like comma-separated made it feel more conversational.

Reviewer 2: It's fine to leave this as you've suggested @tstromberg - I can't propose any better wording.

{{% /capture %}}

{{% capture steps %}}

## Installing minikube

{{< tabs name="tab_with_md" >}}
{{% tab name="Linux" %}}

### Binary download

> For Linux users, there are 3 easy download options:

Reviewer: this line should be switched with the line above

```shell

>  curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64

Reviewer 2: Why the initial space?

 sudo install minikube-linux-amd64 /usr/local/bin/minikube
```

### Debian package

```shell
curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube_latest_amd64.deb
sudo dpkg -i minikube_latest_amd64.deb
```

### RPM package

```shell
curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-latest.x86_64.rpm
sudo rpm -ivh minikube-latest.x86_64.rpm
```

{{% /tab %}}
{{% tab name="macOS" %}}

> If the [Brew Package Manager](https://brew.sh/) installed:

Reviewer: > If the [Brew Package Manager](https://brew.sh/) is installed:

maybe add the word `is`

```shell
brew install minikube
```

Otherwise, download minikube directly:

```shell
curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-darwin-amd64
sudo install minikube-darwin-amd64 /usr/local/bin/minikube
```

{{% /tab %}}
{{% tab name="Windows" %}}

> If the [Chocolatey Package Manager](https://chocolatey.org/) is installed, use it to install minikube:

Reviewer 2: I checked the name for the product. I think this wording works better
```suggestion
If you have the [Chocolatey](https://chocolatey.org/) package manager installed, use it to install minikube:
```
(minor detail, OK to merge without heeding this)

```shell
choco install minikube
```

> Otherwise, download and run the [Windows installer](https://storage.googleapis.com/minikube/releases/latest/minikube-installer.exe)

Reviewer 2: ```suggestion
Otherwise, download and run the [Windows installer](https://storage.googleapis.com/minikube/releases/latest/minikube-installer.exe).
```
(minor detail, OK to merge without heeding this)

{{% capture overview %}}

This page shows you how to install [minikube](https://minikube.sigs.k8s.io/), a tool that runs a Kubernetes cluster on your personal computer.

{{% /capture %}}

{{% capture prerequisites %}}

To run minikube, you'll need:

> * A host with administrator privileges (for example, ability to `sudo` or `Run as Administrator`)

Reviewer 2: How about making this be after the “compatible container or virtual machine manager” point, as a nested item, and explain that if your VMM / container runtime needs you to have admin access then you will need it (for example: VirtualBox and KVM usually don't need you to have superuser access).
If someone does not have this access, I'm concerned that they might see this and incorrectly infer that Minikube does not meet their needs, even if it would in fact work.

I'm not sure if Minikube automatically selects a suitable runtime based on its guess as to whether you have that access. That might be a note to mention if it does not.

Author: As currently written, the installation instructions all require administrative access to install, since they either call `sudo` or interact with a package manager which requires administrative access.

There are edge cases where a user could feasibly install minikube without administrative access, but it would involve straying from these instructions.

{{% capture whatsnext %}}

* [Running Kubernetes locally via minikube](/docs/setup/learning-environment/minikube/)

If you would like additional help to get started, please see:

> 

Reviewer 3: Hi @tstromberg .
Not sure what happened to the order of the content.
I would expect all of the steps to be listed within the `capture steps` directive.
I don't think moving the `help starting minikube` section into the `What's next` section works as well.

Author: I thought the order was kind of weird too. I've addressed it and simplified the text some. PTAL.

{{% /capture %}}

## Starting your first cluster

> From a terminal with administrator access (but not logged in as root), run:

Reviewer 2: Administrator access? That wasn't listed as a prerequisite earlier. Also, I don't think it's always needed.

Author: I'm on the fence on this one. What privileges are necessary is entirely dependent on the hypervisor and host configuration. For instance, Hyper-V by default does need to be run in a terminal as administrator. Docker needs access to a `docker` group. Hyperkit needs `sudo`. 

I removed the verbiage here, but have added the admin access to the list of pre-requisites.

```shell
minikube start
```

> If minikube fails to start, see the [minikube drivers page](https://minikube.sigs.k8s.io/docs/docs/drivers) for help setting up a compatible container or virtual-machine manager.

Reviewer: Should we direct users to create issues in the k/minikube repo? I have seen a few issues created in the k/website repo which then don't get prompt responses. Actually there is a "Create Issue" button at the bottom of the page that leads to k/website repo.

Reviewer 2: We could also point users to Discourse / Slack etc as places to ask for advice. Maybe.

{{% /capture %}}

## Starting your first cluster

Run:

```shell
minikube start
```

> minikube will automatically select an appropriate container or hypervisor driver, but will mention alternatives it available. You may select an alternative driver by using the `--driver` flag. For more information, see the [minikube drivers documentation](https://minikube.sigs.k8s.io/docs/drivers/)

Reviewer 3: Possibly reword `alternatives it`:
During startup, minikube selects an appropriate container or hypervisor driver and informs you
about the availability of alternative containers or drivers.

Reviewer 3: nit: Add a period at end of sentence
For more information, see the [minikube drivers documentation](https://minikube.sigs.k8s.io/docs/drivers/).

What do you think about moving, `If you need help ...` to the What's next section?

Author: Great suggestions. How's this look?

minikube start
```

minikube will automatically select an appropriate driver, but will mention alternatives if available. For additional information or assistance, please see:

* [minikube documentation](https://minikube.sigs.k8s.io/docs)
* [#minikube chat on Slack](https://kubernetes.slack.com/archives/C1F5CT6Q1)
* [minikube issues on GitHub](https://github.com/kubernetes/minikube/issues)

{{% capture whatsnext %}}

> * [Running Kubernetes locally via minikube](/docs/setup/learning-environment/minikube/)

Reviewer: Should we point users to the minikube site? Looks like kind does that too. They just direct you to the kind documentation.

Author: @zacharysarah discussed this for a bit, and we came to the shared conclusion that the best user experience was if the introductory tutorial was hosted here, but all of the reference information was linked to the minikube site.
